llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -2840,6 +2883,9 @@ bool AArch64LoadStoreOpt::optimizeBlock(MachineBasicBlock &MBB,
     updateDefinedRegisters(*MBBI, DefinedInBB, TRI);
     if (TII->isPairableLdStInst(*MBBI) && tryToPairLdStInst(MBBI))
       Modified = true;
+    else if (CanPairFillSpill && isPairableFillSpillInst(*MBBI) &&
+             tryToPairLdStInst(MBBI))
+      Modified = true;
```

Comment:
It is necessary to have a separate path for SVE rather than extending isPairableLdStInst?  I can see isPairableLdStInst is also used to group related instructions together, which is something we might want in the future, assuming it doesn't already come for free.

Reply:
I believe `isPairableLdStInst` is static, but I needed to query the subtarget to perform the checks that `CanPairFillSpill` is doing. I thought it would be better to add the new path to reduce the number of changes, but I'm happy to revisit this if you have a better idea.

Reply:
I've moved these checks into `isPairableLdStInst` and `isCandidateToMergeOrPair` - please let me know what you think of the new approach, otherwise I'm happy to revert it. :)